scripts/smoke-trailbase.py

```
@@ -0,0 +1,1051 @@
+def start_fixture_release(
+    executable: Path,
+    root: Path,
+    port: int,
+    initial_password: list[str] | None = None,
+) -> tuple[subprocess.Popen[bytes], threading.Thread | None]:
+    environment = {key: value for key, value in os.environ.items() if not key.startswith("TRAIL_")}
+    environment["RUST_LOG"] = "info" if initial_password is not None else "warn"
+    output: int | None = subprocess.PIPE if initial_password is not None else subprocess.DEVNULL
+    old_umask = os.umask(0o077)
+    try:
+        process = subprocess.Popen(  # nosec -- nosemgrep -- exact digest-verified fixture; fixed local argv, no shell.
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)